apps/sel4test-tests/src/tests/smc.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2017, Data61, CSIRO (ABN 41 687 119 230)
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <autoconf.h>
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sel4/sel4.h>
+#include <sel4utils/arch/util.h>
+
+#include <vka/object.h>
+#include <vka/capops.h>
+
+#include "../test.h"
+#include "../helpers.h"
+
+#ifdef CONFIG_ALLOW_SMC_CALLS
+
+#define ARM_STD_SVC_VERSION    0x8400ff03
+#define UNASSIGNED_SMC         0x82000000
+
+static int test_smc_calls(env_t env)
+{
+    seL4_ARM_SMCContext smc_args;
+    seL4_ARM_SMCContext smc_results;
+    int error;
+
+    seL4_CPtr smc_cap = env->smc;
+    seL4_CPtr badged_smc_cap = get_free_slot(env);
+
+    error = cnode_mint(env, smc_cap, badged_smc_cap, seL4_AllRights, ARM_STD_SVC_VERSION);
+    test_error_eq(error, seL4_NoError);
+
+    /* Set function and arguments */
+    smc_args.x0 = ARM_STD_SVC_VERSION;
+    smc_args.x1 = 0;
+    smc_args.x2 = 0;
+    smc_args.x3 = 0;
+    smc_args.x4 = 0;
+    smc_args.x5 = 0;
+    smc_args.x6 = 0;
+    smc_args.x7 = 0;
+
+    /* This should succeed */
+    error = seL4_ARM_SMC_Call(badged_smc_cap, &smc_args, &smc_results);
+    test_error_eq(error, seL4_NoError);
+
+    test_eq(smc_results.x0, 0UL);
+    test_eq(smc_results.x1, 1UL);
+    test_eq(smc_results.x2, 0UL);
+    test_eq(smc_results.x3, 0UL);
```

Comment:
Results are put in x0 and x1, so perhaps initialise x2 and x3 to 2 and 3, to check that the kernel doesn't clobber anything? And perhaps make it future proof by checking that (major + minor) >= 0.1?

Reply:
Good ideas. I will do that. Although, major + minor would be 1, so it should be (major + minor) >= 1, correct?

Reply:
Although, that adjustment would pass if `seL4_ARM_SMC_Call()` did absolutely nothing. So I think I should also add a check to make sure that `smc_results.x0 != ARM_STD_SVC_VERSION`.

Reply:
> Results are put in x0 and x1, so perhaps initialize x2 and x3 to 2 and 3, to check that the kernel doesn't clobber anything?

I tried this, but the SMC call clobbers it. 

![image](https://github.com/seL4/sel4test/assets/986572/15468f5b-ddcd-401c-8130-5da294a4e4ba)

We can't actually trust anything in the `seL4_ARM_SMCContext` to be unmodified.